src/transformers/integrations/bitsandbytes.py

```diff
@@ -496,7 +496,9 @@ def _validate_bnb_multi_backend_availability(raise_exception):
                 "You have Intel IPEX installed but if you're intending to use it for CPU, it might not have the right version. Be sure to double check that your PyTorch and IPEX installs are compatible."
             )
 
-        available_devices.discard("cpu")  # Only Intel CPU is supported by BNB at the moment
+        available_devices = frozenset(
+            [device for device in available_devices if device != "cpu"]
+        )  # Only Intel CPU is supported by BNB at the moment
```

Comment:
I'm OK with this for now, especially if it unblocks, but I do want to note that we're intending to have some level of CPU support, x86-64 or not, and with or without IPEX, so this will be changing in the near future..